lib/emerald/grammar/emerald.tt

```diff
@@ -63,25 +63,19 @@ grammar Emerald
     ( variable / !"\n" . )* <InlineLiteral>
   end
 
-  # tag space* string (0 or 1), attr_list (0 or 1)
-  # fix this.
   rule tag_statement
-    tag lparen attributes rparen <TagStatement> /
-    tag space? (( !"\n" . )*)? space* attr_list? <TagStatement>
+    tag space? (( !lparen !newline . )*)? attr_list? <TagStatement>
   end
 
-  # attribute list for
-  # lparen newline (or none), attribute* rparen
   rule attr_list
-    lparen newline attribute* space* rparen <AttributeList>
+    lparen newline lbrace newline attributes rbrace newline rparen <AttributeList>
   end
 
   rule attributes
     attribute* <Attributes>
   end
 
-  # if newline option, must indent
   rule attribute
-    space* attr space* equals space* string comma? <Attribute>
+    attr space* string newline
```

Comment:
I agree with that, but it's the attribute rule - so would that make sense? We currently have variable escaping and multiple lines for the `text_content` rule, so that would mean that something like this is valid:

`text_content`
```
* multiline literal
h1 Header text (
  id ->
    this-is-an-id
       here-is-more-of-the-id
    in-a-multiline-literal
  class this-is-a-class
)
```

The way it currently stands is:

`string`
```
h1 Header text (
  id "this-is-an-id"
  class "this-is-a-class"
)
```

If we change it to a `text_content`, then the first example will be accepted by the grammar. Is that something we want?